### PR TITLE
[Python 2 to 3] Import fixes & small fixes

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- `__cmp__` replaced
- fixed `UserDict` and `Queue` imports
- add `__bool__` alias of `__nonzero__`
- use BytesIO and fix imports of `StringIO`
- bunch of random fixes

### CODE REVIEW

File: test/regression.uts [hunk 1]
```diff
@@ -8034,10 +8039,10 @@ assert(r6 == "d279:1205:e445:5a9f:db28:efc9:afd7:f594")
 
 random.seed(0x2807)
 r6 = RandIP6("2001:db8::-") 
-assert(r6 == "2001:0db8::afd7")
+assert(r6 == "2001:0db8::e445")
```
Comment (gpotter2) [inline test/regression.uts]: Those changes is beause the new `__cmp__` methods calls `_fix()` only once (the legacy one called it between 1-3 times)

**Comment by guedou** [comment]
Looks good ! Could you update CONTRIBUTING with these rules, or at least the BytesIO thing ?

Whats is the meaning of `__bool__ = __nonzero__` ?

**Comment by gpotter2** [comment]
@guedou They suddenly (and silently) decided to rename this:
https://docs.python.org/2/reference/datamodel.html#object.__nonzero__
into this:
https://docs.python.org/3/reference/datamodel.html#object.__bool__

**Comment by guedou** [comment]
.・゜゜・（／。＼）・゜゜・．

**Comment by gpotter2** [comment]
I've updated CONTRIBUTING.md

**Comment by gpotter2** [comment]
@guedou @p-l- This PR can now be merged. Last one will be https://github.com/secdev/scapy/pull/674...

**Pull request merged**